src/gui/editors/SongEditor.cpp

```diff
@@ -837,17 +833,49 @@ void SongEditor::updatePositionLine()
 
 
 
+//! Convert zoom slider's value to bar width in pixels
+int SongEditor::calculatePixelsPerBar() const
+{
+	// What we need to raise 2 by to get MIN_PIXELS_PER_BAR and MAX_PIXELS_PER_BAR
+	static const double minExp = std::log2(MIN_PIXELS_PER_BAR);
+	static const double maxExp = std::log2(MAX_PIXELS_PER_BAR);
+	double exponent = m_zoomingModel->value() / static_cast<double>(ZOOM_STEPS) * (maxExp - minExp) + minExp;
+
+	double ppb = std::pow(2, exponent);
+
+	return static_cast<int>(std::round(ppb));
```

Comment:
- You can replace `std::pow(2, exponent)` with `std::exp2(exponent)`. `log`-`exp` pair is also possible.
- If you're caching `minExp` and `maxExp`, why don't you calculate `1 / static_cast<double>(ZOOM_STEPS) * (maxExp - minExp)` as well?

Reply:
Ok. I think that I've done what you suggested.